miku/src/main/scala/org/http4s/miku/NettyModelConversion.scala

```diff
@@ -181,6 +269,13 @@ object NettyModelConversion {
           new DefaultHttpContent(Unpooled.wrappedBuffer(bytes.toArray))
       }
 
+  private def bytebufToArray(buf: ByteBuf): Array[Byte] = {
+    val array = new Array[Byte](buf.readableBytes())
+    buf.readBytes(array)
+    buf.release()
+    array
+  }
+
   private val CachedEmpty: DefaultHttpContent =
```

Comment:
I'm pretty sure you can `toArray` on a bytebuffer to turn it into an array. It should use `Arrays.copy` or `System.arrayCopy` iirc

Reply:
`ByteBuf` is a netty construct, which you need to explicitly release. That is, imagine `free` using C.

It's not a `ByteBuffer`. If it was and I didn't have to worry about `release()`, I'd just use `Chunk.bytebuffer` but it's not :(.